anta/tests/hardware.py

```diff
@@ -679,8 +682,11 @@ def test(self) -> None:
 
     def _verify_all_slots_populated(self, inventory: dict[str, dict[str, Any]]) -> None:
         """Verify that all available slots for all component types are populated."""
-        for component_data in inventory.values():
-            self._report_failures(component_data)
+        for component_type, component_data in inventory.items():
+            # atomic results
+            result = self.result.add(description=f"All available slots for {component_type.replace('_', ' ').title()}")
+            result.is_success()
```

Comment:
Please rebase with main and use the `status` argument everywhere.

Also, we should have one atomic result per component type with the same description everywhere.
```suggestion
            # Atomic result
            result = self.result.add(description=f"{component_type.replace('_', ' ').title()}", status=AntaTestStatus.SUCCESS)
```